executor/predictor/ready_checker.go

```diff
@@ -2,13 +2,30 @@ package predictor
 
 import (
 	"fmt"
+	"github.com/seldonio/seldon-core/executor/api"
 	"github.com/seldonio/seldon-core/operator/apis/machinelearning.seldon.io/v1"
 	"net"
+	"net/http"
+	"net/url"
+	"strconv"
 )
 
-func Ready(node *v1.PredictiveUnit) error {
+func Ready(protocol string, node *v1.PredictiveUnit) error {
+	switch protocol {
+	case api.ProtocolSeldon:
+		return ReadyHealth(node, "/api/v1.0/health/status")
```

Comment:
It seems that this URL won't work as it requires to be implemented (like the predict method) otherwise it errors out. A good solution would be to add a default behaviour to the funciton if the implementation is not found, that way we could also allow users to extend this which would be quite useful. The alternative (quick workaround) would be to use `/health/ping` instead, which would always return response, but it would not allow for users to extend it

Reply:
Actually it may be best to use `/health/ping` as this would ensure it has backwards compatibility, perhaps we could make the URL for the v1 server configurable via env var as well so that it's also possible to allow users to change the path if they do want to add custom extension

Reply:
It looks ok to me https://github.com/SeldonIO/seldon-core/blob/fd63953ea7e47048eebe5cff148c0053e90c4c7b/python/seldon_core/user_model.py#L512-L529

Reply:
Yeah you're right, I think it may just be that the tests are actually inheriting from the base class (which although mentioned in the docs it's not really common practice), which would basically throw an exception if not implemented

https://github.com/SeldonIO/seldon-core/blob/fd63953ea7e47048eebe5cff148c0053e90c4c7b/python/seldon_core/user_model.py#L140-L143

Reply:
This seems to reflect the errors in the tests ```2022-04-04 19:19:37,408 - seldon_core.wrapper:handle_invalid_usage:60 - ERROR:  {'status': {'status': 1, 'info': 'health is not implemented', 'code': -1, 'reason': 'MICROSERVICE_BAD_DATA'}}```